### PR TITLE
Add new solutions to fibonacci and factorial 

### PR DESCRIPTION
**Pull request by AndreLucasrs**
Add a new iterative fibonacci solution and a recursion factorial solution

### CODE REVIEW

File: src/main/java/math/FibonacciSnippet.java [hunk 1]
```diff
@@ -42,4 +45,20 @@ public static int fibonacci(int n) {
...
     }
   }
+
+  /**
+   * Example of what an iterative implementation of Fibonacci looks like.
+   *
+   * @param number given number
+   * @return fibonacci number for given n
+   */
+  public static int interactiveFibonacci(int number) {
```
Comment (iluwatar) [inline src/main/java/math/FibonacciSnippet.java:55]: Shouldn't the method be called `iterativeFibonacci`?

**Review by iluwatar — COMMENTED** [review]
Looks good, just one comment related to the naming.

File: src/main/java/math/FibonacciSnippet.java (hunk 1 above)
Comment (AndreLucasrs) [reply src/main/java/math/FibonacciSnippet.java]: I ended up not realizing the error, adjusted

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Thank you @AndreLucasrs, this looks great.

**Comment by AndreLucasrs** [comment]
Could you add the Hacktoberfest information ?
because it was not accounted for

**Comment by iluwatar** [comment]
The repository itself has the `hacktoberfest` label

**Pull request merged**